### PR TITLE
Add ray.is_initialized() function.

### PR DESCRIPTION
**Pull request by robertnishihara**
Add `ray.is_initialized()` which checks if `ray.init()` has been called yet.

### CODE REVIEW

**Comment by robertnishihara** [comment]
cc @marianogabitto

**Review by raulchen — APPROVED** [review]
one minor comment, otherwise looks good.

File: test/runtest.py [hunk 1]
```diff
@@ -2536,5 +2536,20 @@ def f():
         ray.experimental.flush_evicted_objects_unsafe()
 
 
+class InitializationTest(unittest.TestCase):
+    def tearDown(self):
+        ray.shutdown()
```
[the remark this answers is not in the document]
Comment (robertnishihara) [reply test/runtest.py]: Good point, I fixed this.

**Comment by robertnishihara** [comment]
jenkins, retest this please

**Comment by raulchen** [comment]
jenkins, retest this please

**Comment by robertnishihara** [comment]
jenkins, retest this please

**Comment by robertnishihara** [comment]
@raulchen I think the Jenkins failure is unrelated to this change.

**Pull request merged**